docs/reference/mapping/fields/routing-field.asciidoc

```diff
@@ -109,3 +109,28 @@ documents with the same `_id` might end up on different shards if indexed with
 different `_routing` values.
 
 It is up to the user to ensure that IDs are unique across the index.
+
+==== Routing to an index partition
+
+An index can be configured such that custom routing values will go to a subset of the shards rather
+than a single shard. This helps mitigate the risk of ending up with an imbalanced cluster while still
+reducing the impact of searches.
+
+This is done by providing the index level setting `index.routing_partition_size` at index creation.
```

Comment:
Link to the setting details:

    <<routing-partition-size,`index.routing_partition_size`>>